ingestion_server/ingestion_server/cleanup.py

```diff
@@ -93,7 +113,82 @@ def _cleanup_tags(tags):
 }
 
 
-def _clean_data_worker(rows, temp_table, providers_config):
+class TlsTest:
+    """
+    URLs crawled from upstream are often lacking protocol information, or
+    use HTTP when HTTPS is available. We have to test a small sample of the
+    URLs to determine what protocol should be appended to each URL in the
+    event that it is missing or incorrect.
+    """
+    @classmethod
+    def _test_tls_supported(cls, url):
+        # No protocol provided
+        if 'https://' not in url and 'http://' not in url:
+            fixed_url = 'http://' + url
+            return cls._test_tls_supported(fixed_url)
+        # HTTP provided, but we want to check if HTTPS is supported as well.
+        elif 'http://' in url:
+            https = url.replace('http://', 'https://')
+            try:
+                res = re.get(https, timeout=2)
+                log.info('{}:{}'.format(https, res.status_code))
+                return 200 <= res.status_code < 400
+            except re.RequestException:
+                return False
+        # If HTTPS is in the URL already, we're going to trust that HTTPS is
+        # supported.
+        else:
```

Comment:
nitpick: you don't need the else here, you can just `return True` outside the conditional